### PR TITLE
Improved notebook cell drag images

### PR DESCRIPTION
**Pull request by jonah-iden**
#### What it does

This improves the rendering of drag images for notebook cells and aligns to more how they look in vscode.

#### How to test

Open notebook and try dragging a cell. 
Code and mardown cells have different drag image renderers

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/notebook/src/browser/view/notebook-code-cell-view.tsx [hunk 1]
```diff
@@ -102,7 +107,20 @@ export class NotebookCodeCellRenderer implements CellRenderer {
...
+        dragImage.style.height = '100px';
+        dragImage.style.display = 'flex';
+
+        const fakeRunButton = document.createElement('span');
+        fakeRunButton.className = `${codicon('play')} theia-notebook-cell-status-item`;
+        dragImage.appendChild(fakeRunButton);
+
+        const fakeEditor = document.createElement('div');
+        dragImage.appendChild(fakeEditor);
+        const firstLine = new MarkdownString(`\`\`\`${cell.language}\n${cell.source.split('\n')[0]}\n\`\`\``, { supportHtml: true, isTrusted: false });
```
Comment (msujew) [inline packages/notebook/src/browser/view/notebook-code-cell-view.tsx:120]: Suggestion: Using 100 px makes this perfect to show the first 5 lines. IMO this looks way better:
image
```suggestion
        const lines = cell.source.split('\n').slice(0, 5).join('\n');
        const markdownCode = new MarkdownString(`\`\`\`${cell.language}\n${lines}\n\`\`\``, { supportHtml: true, isTrusted: false });
```

File: packages/notebook/src/browser/view/notebook-cell-list-view.tsx [hunk 2]
```diff
@@ -137,14 +137,14 @@ export class NotebookCellListView extends React.Component<CellListProps, Noteboo
             return;
         }
 
-        if (this.dragGhost) {
-            this.dragGhost.remove();
+        if (NotebookCellListView.dragGhost) {
+            NotebookCellListView.dragGhost.remove();
```
Comment (msujew) [inline packages/notebook/src/browser/view/notebook-cell-list-view.tsx:141]: Minor:
```suggestion
        NotebookCellListView.dragGhost?.remove();
```

**Review by msujew — COMMENTED** [review]
I'm wondering whether we should call `NotebookCellListView.dragGhost?.remove()` on `onDrop` as well. Otherwise the element stays in the `body` indefinitely, right?

Aside from that, looks pretty great. Good idea using the markdown renderer to produce "pseudo" editors during dragging.

**Review by msujew — COMMENTED** [review]
I've managed to break the markdown rendering by including ```` ``` ```` in the first line of the cell source:

image

I think it would be alright to just run `replaceAll('```', '')` on the `lines` value. It probably doesn't matter if it had any significance in the input text.

Aside from that, I've noticed that scrolling in the notebook doesn't work while dragging a cell, but that's unrelated to the PR.

**Comment by jonah-iden** [comment]
@msujew thanks, good catch. fixed it the way you suggested for. Not super happy with that though since in python iirc you use ``` to indicate multi line strings.

The scrolling seems to be general theia problem. Same thing happens in the file explorer and other places as soon as you drag anything. If there is not already an issue for that we should probably reate one

**Comment by msujew** [comment]
@jonah-iden AFAIK python uses `"""`. Anyway, I found a way to correctly escape everything and pushed it to your branch.

**Review by msujew — APPROVED** [review]
Looks good to me, thanks 👍

**Pull request merged**